expansion.bzl

```diff
@@ -0,0 +1,561 @@
+"""
+Module containing functions that aid in environment variable expansion.
+
+This could be replaced with the same impl via Skylib dependency, if/when PR is merged.
+See https://github.com/bazelbuild/bazel-skylib/pull/486.
+"""
+
+_CONSIDERED_KEY_FORMATS = ("${}", "${{{}}}", "$({})")
+
+def _valid_char_for_env_var_name(char):
+    return char.isalnum() or char == "_"
+
+def _find_env_var_name_index_index(
+        string,
+        str_len,
+        search_start,
+        special_ending_char = None):
+    for offset in range(str_len - search_start):
+        index = search_start + offset
+        char = string[index]
+        if special_ending_char:
+            if char == special_ending_char:
+                return index
+        elif not _valid_char_for_env_var_name(char):
+            return index - 1
+    return str_len - 1
+
+def _odd_count_dollar_sign_repeat(containing_str, end_of_dollar_signs_index):
```

Comment:
This function appears more in the `not odd` context, so perhaps flip? Or something...

Reply:
Sure. Switched to `_even_count_dollar_sign_repeat()`. Updated logic at call sites.